File: src/main/java/com/nuutrai/reactor/util/Settable.java
package com.nuutrai.reactor.util;

public class Settable<T> {

    private T value;

    public void set(T value) {
        this.value = value;
    }

    public T get() {
        return value;
    }

    public boolean isNull() {
        return value == null;
    }
}

File: src/main/kotlin/com/nuutrai/reactor/util/Settable.kt
package com.nuutrai.reactor.util

class Settable<T> {
	private var value: T? = null

	fun set(value: T?) {
		this.value = value
	}

	fun get(): T? {
		return value
	}

	val isNull: Boolean
		get() = value == null
}
